docs/getting-started/auth.adoc

```diff
@@ -1,15 +1,169 @@
+* Specify the schema of the redirect url for your Android project in `android.defaultConfig.manifestPlaceholders` of the apps `build.gradle` file.
+It should match the schema specified in the `Valid Redirect URL` section for the client in Keycloak.
+It is recommended to use the package name of the Android app as the schema of the redirect url to avoid conflicts.
+
+=== Initializing the SDK
+
+`AuthService` can be retrieved using the `MobileCore#getInstance` method.
+
+[source,java]
+----
+AuthService authService = mobileCore.getInstance(AuthService.class);
+----
+
+Any subsequent `MobileCore#getInstance` method will return the same instance of `AuthService`.
+
+Before the `AuthService` can be used used `AuthService#init` must be invoked once in an app.
+
+If `AuthService#init` is not invoked then an `IllegalStateException` will be thrown when using the
+service.
+
+[source,java]
+----
+AuthServiceConfiguration authServiceConfig = new AuthServiceConfiguration.AuthConfigurationBuilder()
+    .withRedirectUri("org.aerogear.mobile.example:/callback")
+    .build();
+
+// Only invoke this once, every subsequent retrieval of the AuthService will retrieve the same, already initialized, instance.
+authService.init(context, authServiceConfig);
+----
+
+=== Retrieving the current user
+
+To retrieve the current authenticated user the `AuthService#currentUser` method can be invoked. This will be `null` if there is
+no user authenticated. So it can be used to check whether to start the authentication flow or not.
+
+[source,java]
+----
+// authService already initialized.
+UserPrincipal currentUser = authService.currentUser();
+
+if (currentUser) {
+    // User is authenticated, get the users name
+    String userName = currentUser.getName();
+    // Get the users email address
+    String userEmail = currentUser.getEmail();
+    // Get the access token of the authenticated user
+    String accessToken = currentUser.getAccessToken();
+} else {
+    // User is not authenticates, start authentication flow
+}
+----
+
+=== Authenticating
+
+To start the authentication invoke the `AuthService#login` method.
+
+[source,java]
+----
+// authService already initialized.
+AuthService authService = mobileCore.getInstance(AuthService.class);
+
+static int LOGIN_RESULT_CODE = 1;
+Activity myActivity = this.getActivity();
+
+// Build the options object and start the authentication flow.
+OIDCAuthenticateOptions options = new OIDCAuthenticateOptions(myActivity, LOGIN_RESULT_CODE);
+
+Callback authCallback = new Callback<UserPrincipal>() {
+    @Override
+    public void onSuccess(UserPrincipal principal) {
+        // User authenticated in, continue on..
+    }
+
+    @Override
+    public void onError(Throwable error) {
+        // An error occurred during login.
+    }
+}
+
+authService.login(options, authCallback);
+----
+
+Once the browser returns to the app the result can be handled. In the activity provided to the
```

Comment:
Can we add a code example for what's meant to be done here?

I'm guessing its something like this in my Activity
```java
    @Override
    public void onActivityResult() {
        Intent intent = new Intent();
        authService.handleAuthResult();
    }
```

but that doesn't seem to be an overriden method. Was I suppose to 'implement' some extra interface?
Also, what exactly am I suppose to do in this method? What if I don't override it?